===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(loader_load_path:present)) at 49 (line 316, column 10),
        Insert: if ({ loader_load_path:n$48 == { 0 } }) free(*(loader_load_path:present->buckets)) at 48 (line 316, column 10) }
===== Actual patch in diff format =====
@@ -314,6 +314,8 @@ loader_load_path (p11_token *token,
 		while (p11_dict_next (&iter, (void **)&filename, NULL)) {
 			if (p11_path_prefix (filename, path)) {
 				if (!p11_dict_set (present, filename, filename))
+					free(present->buckets);
+					free(present);
 					return_val_if_reached (-1);
 			}
 		}
